fontforge/splinestroke.c

```diff
@@ -3129,11 +3129,11 @@ return( NULL );
 		int isnext=sp->next->to==sp2, isprev=sp->prev->from==sp2;
 		BreakLine(sp->next,start,end,&first1,&second1);
 		BreakLine(sp2->next,end,start,&first2,&second2);
-		if ( first1->me.x!=second2->me.x || first1->me.y!=second2->me.y ) {
+		if ( first1->me.x!=second2->me.x || first1->me.y!=second2->me.y || first1==second2 ) {
```

Comment:
Okay. I wasn't sure how to handle that case, because if you look at the code block following this, `first1` gets assigned some values from `second2` and `second1` some from `first2`. Then `first2` and `second2` get freed - so you see that if `first2==second1` or `second2=first`, we have a problem, because they've already been freed.